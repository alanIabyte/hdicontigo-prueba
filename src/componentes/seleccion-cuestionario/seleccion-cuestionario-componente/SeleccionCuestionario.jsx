import React from "react";
import PropTypes from "prop-types";
import {
  CuerpoSeleccionCuestionario,
  PreguntaCuestionario,
  RespuestaCuestionario,
  SeleccionSiCuestionario,
  SeleccionNoCuestionario,
  CampoRequeridoSeleccionCuestionario,
} from "./SeleccionCuestionario.styled";

const SeleccionCuestionario = (props) => {
  const { cambiarEstado, pregunta, respuesta, mostrarMensajeCampoRequerido } =
    props;
  const diccionario = {
    respuestaAfirmativa: "SÃ­",
    respuestaNegativa: "No",
    mensajeCampoRequerido: "Campo requerido para poder continuar",
  };

  return (
    <CuerpoSeleccionCuestionario id="componentePregunta">
      <PreguntaCuestionario id="textoPregunta">
        {pregunta}
        <RespuestaCuestionario>
          <SeleccionSiCuestionario
            id="botonSi"
            onClick={() => {
              cambiarEstado(true);
            }}
            type="submit"
            activado={respuesta}
          >
            {diccionario.respuestaAfirmativa}
          </SeleccionSiCuestionario>
          <SeleccionNoCuestionario
            id="botonNo"
            onClick={() => {
              cambiarEstado(false);
            }}
            type="submit"
            activado={respuesta}
          >
            {diccionario.respuestaNegativa}
          </SeleccionNoCuestionario>
        </RespuestaCuestionario>
      </PreguntaCuestionario>
      <CampoRequeridoSeleccionCuestionario
        id="error"
        mostrar={mostrarMensajeCampoRequerido}
      >
        {diccionario.mensajeCampoRequerido}
      </CampoRequeridoSeleccionCuestionario>
    </CuerpoSeleccionCuestionario>
  );
};

SeleccionCuestionario.propTypes = {
  cambiarEstado: PropTypes.func,
  mostrarMensajeCampoRequerido: PropTypes.bool,
  pregunta: PropTypes.string,
  respuesta: PropTypes.bool,
};

SeleccionCuestionario.defaultProps = {
  pregunta: "",
  mostrarMensajeCampoRequerido: null,
  respuesta: null,
  cambiarEstado() {},
};

export default SeleccionCuestionario;
