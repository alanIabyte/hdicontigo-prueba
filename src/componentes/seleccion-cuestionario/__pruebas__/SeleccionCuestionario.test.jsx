/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import renderer from "react-test-renderer";
import SeleccionCuestionario from "../seleccion-cuestionario-componente/SeleccionCuestionario";
import {
  SeleccionCuestionarioVacio,
  SeleccionCuestionarioConRespuesta,
} from "../__simulaciones__/SeleccionCuestionario.simulacion";

describe("#Seleccion Cuestionario", () => {
  it("se despliega correctamente sin respuesta", () => {
    const tree = renderer
      .create(<SeleccionCuestionario {...SeleccionCuestionarioVacio} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("se despliega correctamente con respuesta", () => {
    const tree = renderer
      .create(<SeleccionCuestionario {...SeleccionCuestionarioConRespuesta} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
