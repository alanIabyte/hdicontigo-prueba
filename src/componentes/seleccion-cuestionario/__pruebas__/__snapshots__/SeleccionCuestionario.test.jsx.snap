// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Seleccion Cuestionario se despliega correctamente con respuesta 1`] = `
<div
  className="sc-ikkxIA"
  id="componentePregunta"
>
  <div
    className="sc-imWYAI sc-dAbbOL eYGTZs deLeNf"
    id="textoPregunta"
  >
    ¿Hay otro vehículo involucrado u otros bienes afectados?
    <div
      className="sc-imWYAI sc-feUZmu eYGTZs gFbaoE"
    >
      <button
        className="sc-fUnMCh dmeRBQ"
        id="botonSi"
        onClick={[Function]}
        type="submit"
      >
        Sí
      </button>
      <button
        className="sc-hzhJZQ kbFIPs"
        id="botonNo"
        onClick={[Function]}
        type="submit"
      >
        No
      </button>
    </div>
  </div>
  <div
    className="sc-eldPxv sc-fHjqPf jeNqaZ jiAKDD"
    id="error"
  >
    Campo requerido para poder continuar
  </div>
</div>
`;

exports[`#Seleccion Cuestionario se despliega correctamente sin respuesta 1`] = `
<div
  className="sc-ikkxIA"
  id="componentePregunta"
>
  <div
    className="sc-imWYAI sc-dAbbOL eYGTZs deLeNf"
    id="textoPregunta"
  >
    ¿Hay personas lesionadas?
    <div
      className="sc-imWYAI sc-feUZmu eYGTZs gFbaoE"
    >
      <button
        className="sc-fUnMCh UPsbs"
        id="botonSi"
        onClick={[Function]}
        type="submit"
      >
        Sí
      </button>
      <button
        className="sc-hzhJZQ kbFIPs"
        id="botonNo"
        onClick={[Function]}
        type="submit"
      >
        No
      </button>
    </div>
  </div>
  <div
    className="sc-eldPxv sc-fHjqPf jeNqaZ jiAKDD"
    id="error"
  >
    Campo requerido para poder continuar
  </div>
</div>
`;
